### PR TITLE
Class5 tic tac toe

### PR DESCRIPTION
**Pull request by tysweet**
Tic Tac Toe!!

### CODE REVIEW

File: 02week/ticTacToe.js [hunk 1]
```diff
@@ -24,23 +26,65 @@ function printBoard() {
 }
 
 function horizontalWin() {
-  // Your code here
+  if ((board[0][0] === 'X') && (board[0][1] === 'X') && (board[0][2] === 'X')) {
```
Comment (reneemeyer) [inline 02week/ticTacToe.js]: You're repeating yourself A LOT here. You don't need to check for both X and O, just player turn

File: 02week/ticTacToe.js [hunk 2]
```diff
@@ -24,23 +26,65 @@ function printBoard() {
...
+    return true;
+  } else if ((board[2][0] === 'O') && (board[2][1] === 'O') && (board[2][2] === 'O')) {
+    return true;
+  }
 }
 
+
 function verticalWin() {
-  // Your code here
+  if ((board[0][0] === 'X') && (board[1][0] === 'X') && (board[2][0] === 'X')) {
```
Comment (reneemeyer) [inline 02week/ticTacToe.js]: You're repeating yourself A LOT here. You don't need to check for both X and O, just player turn

File: 02week/ticTacToe.js [hunk 3]
```diff
@@ -24,23 +26,65 @@ function printBoard() {
...
+  } else if ((board[0][1] === 'O') && (board[1][1] === 'O') && (board[2][1] === 'O')) {
+    return true;
+  } else if ((board[0][2] === 'O') && (board[1][2] === 'O') && (board[2][2] === 'O')) {
+    return true;
+  }
 }
 
 function diagonalWin() {
-  // Your code here
+  if ((board[0][0] === 'X') && (board[1][1] === 'X') && (board[2][2] === 'X')) {
```
Comment (reneemeyer) [inline 02week/ticTacToe.js]: You're repeating yourself A LOT here. You don't need to check for both X and O, just player turn

File: 02week/ticTacToe.js [hunk 4]
```diff
@@ -24,23 +26,65 @@ function printBoard() {
...
+  } else if ((board[0][0] === 'O') && (board[1][1] === 'O') && (board[2][2] === 'O')) {
+    return true;
+  } else if ((board[0][2] === 'O') && (board[1][1] === 'O') && (board[2][2] === 'O')) {
+    return true;
+  }
 }
 
 function checkForWin() {
-  // Your code here
+  horizontalWin();
```
Comment (reneemeyer) [inline 02week/ticTacToe.js]: this should be wrapped in an if statement. you want to ONLY return true if one of these functions return true

File: 02week/ticTacToe.js [hunk 5]
```diff
@@ -24,23 +26,65 @@ function printBoard() {
...
 
 function ticTacToe(row, column) {
-  // Your code here
+  board[row][column] = playerTurn;
+  if (playerTurn === 'X') {
+    (playerTurn = 'O');
+  } else {
+    (playerTurn = 'X');
+  }
+  checkForWin();
```
Comment (reneemeyer) [inline 02week/ticTacToe.js]: You should only switch the turn if there is no win

**Review by reneemeyer — CHANGES_REQUESTED** [review]
Ty good start. Make sure you're playing the game to make sure all of your check for win functions are called correctly.

File: 02week/ticTacToe.js (hunk 1 above)
Comment (tysweet) [reply 02week/ticTacToe.js]: Fixed.

File: 02week/ticTacToe.js (hunk 2 above)
Comment (tysweet) [reply 02week/ticTacToe.js]: Fixed.

File: 02week/ticTacToe.js (hunk 3 above)
Comment (tysweet) [reply 02week/ticTacToe.js]: Fixed.

File: 02week/ticTacToe.js (hunk 4 above)
Comment (tysweet) [reply 02week/ticTacToe.js]: Fixed.

File: 02week/ticTacToe.js (hunk 5 above)
Comment (tysweet) [reply 02week/ticTacToe.js]: Fixed.

**Pull request merged**